File: src/server/src/main/java/com/project/domain/valueobjects/Weight.java
package com.project.domain.valueobjects;

import com.project.common.enums.WeightType;
import jakarta.persistence.Embeddable;
import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;

@Embeddable
public class Weight {
    private double weight;
    @Enumerated(EnumType.STRING)
    private WeightType weightType;

    private Weight(double weight) {
        this.weight = weight;
        this.weightType = WeightType.KG;
    }

    protected Weight() {}

    public static Weight from(int weight) {
        return new Weight(weight);
    }

    public double getWeight() {
        return weight;
    }

    public WeightType getWeightType() {
        return weightType;
    }
}

File: src/server/src/main/kotlin/com/project/domain/valueobjects/Weight.kt
package com.project.domain.valueobjects

import com.project.common.enums.WeightType
import jakarta.persistence.Embeddable
import jakarta.persistence.EnumType
import jakarta.persistence.Enumerated

@Embeddable
class Weight {
    var weight: Double? = null
        private set

    @Enumerated(EnumType.STRING)
    val weightType: WeightType = WeightType.KG

    constructor(weight: Double) {
        this.weight = weight
    }

    protected constructor()
}
